fixes concurrent build issue

hrishin opened the pull request:
- changed the plugin code to manage build throttling configuration from "Manage Jenkins" (central) pane
- added a jobs listener to add throttling behavior while creating new workflow jobs

Fixes https://github.com/openshiftio/openshift.io/issues/2729

rohanKanojia reviewed (approved):
lgtm

lordofthejars commented on src/test/java/hudson/plugins/buildblocker/PipelineJobListenerTest.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Test
+    public void should_add_blocking_properties_from_config_file() throws IOException {
+        FreeStyleProject job1 = jenkins.createFreeStyleProject("testJob");
```
maybe I will call `testJob` instead of `job1`

hrishin replied:
better, fixed it.

lordofthejars commented on src/main/java/hudson/plugins/buildblocker/PipelineJobListener.java:
```diff
@@ -0,0 +1,48 @@
+        if(job.getProperty(BuildBlockerProperty.class) == null) {
+            try {
+                BuildBlockerProperty buildBlockerProperty = BuildBlockerProperty.getDefaultProperties();
+                LOG.log(Level.INFO, "Adding blocking properties "+buildBlockerProperty+" to job "+ job.getName());
```
Stupid thing, can you add white space between `+` of string concatenation? Just that is the de-facto format way :)

lordofthejars commented on src/test/java/hudson/plugins/buildblocker/BuildBlockerQueueTaskDispatcherTest.java:
```diff
@@ -150,6 +142,15 @@ public void testMultipleExecutors() throws Exception {
+            job.removeProperty(BuildBlockerProperty.class);
+            job.addProperty(property);
+        } catch (IOException e) {
+
```
should we add a log message or something? Just to be sure that we do not loose any important information.

hrishin replied:
fixed

hrishin committed:
Limits the concurrent build
- changed the plugin code to manage build throttling configuration from 'Manage Jenkins Configuration' (central) pane
- added the job listener to add throttling behavior for new workflow jobs
- added unit tests to check build blocking properties
- updated README about plugin configuration for fabric8 jenkins deployment
- fixed review comments

Commit:
Merge branch 'master' into iss#2729-throttle
